consensus/polybft/consensus_runtime.go

```diff
@@ -1199,6 +1182,57 @@ func (c *consensusRuntime) BuildRoundChangeMessage(
 	return signedMsg
 }
 
+// getFirstBlockOfEpoch returns the first block of epoch in which provided header resides
+func (c *consensusRuntime) getFirstBlockOfEpoch(epochNumber uint64, latestHeader *types.Header) (uint64, error) {
+	if latestHeader.Number == 0 {
+		// if we are starting the chain, we know that the first block is block 1
+		return 1, nil
+	}
+
+	blockHeader := latestHeader
+	blockExtra, err := GetIbftExtra(latestHeader.ExtraData)
+
```

Comment:
```suggestion
```

Reply:
Linter gets unhappy if I rewmove this :confused: : 

> only one cuddle assignment allowed before if statement (wsl)go-golangci-lint

Reply:
Then cuddle with the next line and separate from the previous, I find it more clean.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/897/commits/b28485cc0c255875b9603276f1a6cbb4bacaf70e